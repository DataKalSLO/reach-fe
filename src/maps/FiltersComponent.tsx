import React from 'react';

import Button from '@material-ui/core/Button';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
import MaterialUIPickers from '../common/components/DatePicker';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'left',
      '& > *': {
        margin: theme.spacing(1)
      }
    }
  })
);

export function DatePickers() {
  const classes = useStyles();
  return (
    <form className={classes.root} noValidate>
      <MaterialUIPickers id="start-picker" label="start-date-picker" />
      <MaterialUIPickers id="end-picker" label="end-date-picker" />
    </form>
  );
}

export default function BasicButtonGroup() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <ButtonGroup color="primary" aria-label="outlined primary button group">
        <Button>Region</Button>
        <Button>City</Button>
        <Button>Zip Code</Button>
      </ButtonGroup>
    </div>
  );
}
