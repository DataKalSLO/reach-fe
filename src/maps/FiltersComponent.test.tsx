import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { GeoFilter } from './FiltersComponent';

React.useLayoutEffect = React.useEffect;

configure({ adapter: new Adapter() });

// Testing basic props of GeoFilter, still do not know how to test clicks and similar events.
describe('<GeoFilter /> without props', () => {
  // supress warnings (mui & jest don't always get along)
  const originalConsoleError = console.error;
  beforeEach(() => {
    console.error = jest.fn(msg => {
      if (msg.includes('Warning: useLayoutEffect does nothing on the server')) {
        return null;
      } else {
        originalConsoleError(msg);
      }
    });
  });
  afterEach(() => {
    console.error = originalConsoleError;
  });
  const props = {
    boundSelection: 'Zip Code' as 'Zip Code',
    setBoundSelection: jest.fn().mockResolvedValue([])
  };
  const container = shallow(<GeoFilter {...props} />);

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});

// Tried to write a test for clicks but had trouble with .find and finding the button
// describe('Testing handleClick() gets called on click', () => {
//   const props = {
//     setBoundSelection: '',
//     boundSelection: ''
//   };

//   it('handleClick should get called', () => {
//     const wrapper = shallow(<GeoFilter {...props} />);
//     wrapper
//       .find('ButtonGroup')
//       .find('button')
//       .simulate('click');

//   });
// });
