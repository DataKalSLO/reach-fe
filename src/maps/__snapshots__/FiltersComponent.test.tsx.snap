// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GeoFilter /> without props should match the snapshot 1`] = `"<div class=\\"makeStyles-root-1\\"><div role=\\"group\\" class=\\"MuiButtonGroup-root\\" aria-label=\\"outlined primary button group\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-outlinedPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Region</span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-outlinedPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Communities</span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-outlinedPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Zip Code</span></button></div></div>"`;
